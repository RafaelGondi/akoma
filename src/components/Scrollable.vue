<template>
	<span
		id="scrollable"
	>
		<div
			class="scrollable__container"
			:style="{
				'max-height': maxHeight,
				'height': maxHeight,
			}"
		>
			<!-- @slot Slot com o conteúdo que você deseja que seja scrollable -->
			<slot></slot>
		</div>
	</span>
</template>

<script>
export default {
	props: {
		/**
		 * Altura máxima do container até ser scrollado
		 */
		maxHeight: {
			type: String,
			default: '300px',
			required: true,
		},
	},
};
</script>
<style lang="scss" scoped>
@import '../assets/sass/app.scss';

#scrollable .scrollable {
	&__container {
		overflow-y: auto;
		/* width */
		&::-webkit-scrollbar {
			width: 6px;
			border-radius: 8px;
		}

		/* Track */
		&::-webkit-scrollbar-track {
			background: transparent;
		}

		/* Handle */
		&::-webkit-scrollbar-thumb {
			background: $cinza-4;
			border-radius: 8px;
		}

		/* Handle on hover */
		&::-webkit-scrollbar-thumb:hover {
			background: $cinza-5;
		}
	}

}
</style>
