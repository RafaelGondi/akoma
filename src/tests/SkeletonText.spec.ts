import { describe, test, expect } from 'vitest';
import Scrollable from '../components/Scrollable.vue';
import { mount } from '@vue/test-utils';

describe('Scrollable', () => {
	test('renders correctly', async () => {
		const wrapper = mount(Scrollable, {
			props: {
				width: '300',
				height: '20',
			},
		});

		expect(wrapper).toMatchSnapshot();
	});
});
