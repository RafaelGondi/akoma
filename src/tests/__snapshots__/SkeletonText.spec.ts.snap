// Vitest Snapshot v1

exports[`SkeletonText > renders correctly 1`] = `"<div data-v-58efc24d=\\"\\" class=\\"skeleton__container\\" style=\\"--width: 300px; --height: 20px;\\"></div>"`;
